test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -913,6 +909,76 @@ protected final BytesReference toShuffledXContent(ToXContent toXContent, XConten
     }
 
     /**
+     * This method takes the input xContent data and adds a random field value, inner object or array into each
+     * json object. This can e.g. be used to test if parsers that handle the resulting xContent can handle the
+     * augmented xContent correctly, for example when testing lenient parsing.
+     *
+     * If the xContent output contains objects that should be skipped of such treatment, an optional filtering
+     * {@link Predicate} can be supplied that checks xContent paths that should be excluded from this treatment.
+     *
+     * This predicate should check the xContent path that we want to insert to and return <tt>true</tt> if the
+     * path should be excluded. Paths are string concatenating field names and array indices, so e.g. in:
+     *
+     * <pre>
+     * {
+     *      "foo1 : {
+     *          "bar" : [
+     *              { ... },
+     *              { ... },
+     *              {
+     *                  "baz" : {
+     *                      // insert here
+     *                  }
+     *              }
+     *          ]
+     *      }
+     * }
+     * </pre>
+     *
+     * "foo1.bar.2.baz" would point to the desired insert location.
+     *
+     * To exclude inserting into the "foo1" object we would user a {@link Predicate} like
+     * <pre>
+     * {@code
+     *      (path) -> path.endsWith("foo1")
+     * }
+     * </pre>
+     *
+     * or if we don't want any random insertions in the "foo1" tree we could use
+     * <pre>
+     * {@code
+     *      (path) -> path.contains("foo1")
+     * }
+     * </pre>
+     */
+    public BytesReference insertRandomFields(XContentType contentType, BytesReference xContent, Predicate<String> excludeFilter)
+            throws IOException {
+        List<String> insertPaths = new ArrayList<>();
+        if (excludeFilter == null) {
+            excludeFilter = (path -> false);
+        }
+        try (XContentParser parser = createParser(contentType.xContent(), xContent)) {
+            List<String> possiblePaths = XContentTestUtils.getInsertPaths(parser);
+            possiblePaths.stream().filter(excludeFilter.negate()).forEach(insertPaths::add);
+        }
+        try (XContentParser parser = createParser(contentType.xContent(), xContent)) {
+            Supplier<Object> value = () -> {
+                if (randomBoolean()) {
+                    return randomAlphaOfLength(10);
```

Comment:
Yes, I think I can use RandomObjects.randomStoredFieldValues() which gives us Strings, all kinds of numbers, boolean or a byte[]. Maybe this will make test failures a bit harder to read but since we expect parsers to skip these values, any randomization here should work (in theory).